### DISCUSSION TITLE
Automatic Invalidate/Reconnect Won't Work Properly

### DISCUSSION THREAD

**Discussion by CapBlackShot** [dbeaver/dbeaver#13684] [Q&A] [answered]
Hi,

I'm using the latest DBeaver version on Windows 10 x64.

I have an active hosting plan that gives me access to a couple of databases. I am able to connect to them just fine, but apparently the hosting company disconnects me automatically after 30s or so.

DBeaver's "auto-recover" feature doesn't seem to be working in such circumstances. I have to manually click invalidate/reconnect in order for it to work.

**Comment by uslss** [comment, marked as answer]
You should set keep-alive option to a not-null value. E/g/ if you set it to 20 then connection will be invalidated every 20 seconds
![image](https://user-images.githubusercontent.com/31996417/128146683-f8483caf-dbd3-4322-95e2-46d7cce2374c.png)

**Comment by CapBlackShot** [comment]
> You should set keep-alive option to a not-null value. E/g/ if you set it to 20 then connection will be invalidated every 20 seconds
> ![image](https://user-images.githubusercontent.com/31996417/128146683-f8483caf-dbd3-4322-95e2-46d7cce2374c.png)

Thank you!! I'll give it a shot.

Do you know if hosting companies allow this interval to be changed? Or if it's a setting that I can change manually, so DBeaver doesn't need to invalidate the connection so often?

**Reply by uslss** [reply]
unfortunately, I didn't know if hosting companies allow this interval to be changed
